sdks/python/apache_beam/dataframe/transforms_test.py

```diff
@@ -266,6 +266,17 @@ def check(actual):
           lambda x: {'res': 3 * x}, proxy)
       assert_that(res['res'], equal_to_series(three_series), 'CheckDictOut')
 
+  def test_cat(self):
+    # verify that cat works with a List[Series] sicne this is missing from doctests
+    df = pd.DataFrame({
+        'one': ['A', 'B', 'C'],
+        'two': ['BB', 'CC', 'A'],
+        'three': ['CCC', 'AA', 'B'],
+    })
+    self.run_scenario(df, lambda df: df.two.str.cat([df.three], join='outer'))
+    self.run_scenario(
+        df, lambda df: df.one.str.cat([df.two, df.three], join='outer'))
+
```

Comment:
I've marked this PR WIP because this test does not pass. It yields the following error, that I'm not sure how to diagnose. @robertwb can you provide any insight here?

```
apache_beam/runners/common.py:806: in _invoke_process_per_window                                                                                                                                                                        
    [si[global_window] for si in self.side_inputs]))                                                                                                                                                                                    
apache_beam/runners/common.py:806: in <listcomp>                                                                                                                                                                                        
    [si[global_window] for si in self.side_inputs]))                                                                                                                                                                                    
apache_beam/runners/worker/bundle_processor.py:422: in __getitem__                                                                                                                                                                      
    self._cache[target_window] = self._side_input_data.view_fn(raw_view)                                                                                                                                                                
apache_beam/pvalue.py:395: in <lambda>                                                                                                                                                                                                  
    lambda iterable: from_runtime_iterable(iterable, view_options))                                                                                                                                                                     
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _                                                                                                             
                                                                                                                                                                                                                                        
it = <apache_beam.runners.worker.bundle_processor._StateBackedIterable object at 0x7f3b0e5cf310>                                                                                                                                        
options = {'coder': WindowedValueCoder[FastPrimitivesCoder], 'window_mapping_fn': <function _global_window_mapping_fn at 0x7f3b1b31b710>}
                                                                                                                                                                                                                                        
    @staticmethod                                                                                                   
    def _from_runtime_iterable(it, options):                                                                                                                                                                                            
      head = list(itertools.islice(it, 2))                                                                                                                                                                                              
      if not head:                                                                                                  
        return options.get('default', EmptySideInput())                                                             
      elif len(head) == 1:                                                                                          
        return head[0]                                                                                                                                                                                                                  
      raise ValueError(                                                                                                                                                                                                                 
          'PCollection of size %d with more than one element accessed as a '                                                                                                                                                            
          'singleton view. First two elements encountered are "%s", "%s".' %                                                                                                                                                            
>         (len(head), str(head[0]), str(head[1])))                                                                                                                                                                                      
E     ValueError: PCollection of size 2 with more than one element accessed as a singleton view. First two elements encountered are "<pandas.core.strings.StringMethods object at 0x7f3b0e32ec90>", "<pandas.core.strings.StringMethods 
object at 0x7f3b0e3d7510>". [while running 'DataframeTransform/Eval/[ComputedExpression[cat_Series_139891620772240]]:139891619912080/FlatMap(evaluate)/FlatMap(evaluate)']  
```

Reply:
I think this is because we're trying to partition the `StringMethods` object itself and send it across the stage/shuffle boundary, and that code only expects dataframes and series to be partition-able (expecting everything else to be a single-object scalar. 

One route would be to add StringMethods to the list of things that are partitionable, but that feels a bit odd and won't work well with a columnar backend representation (e.g. how would we represent s.str[:] as an Arrow batch?) Better would be for these expressions to take the underlying series as an argument, and invoke `str` on it themselves as part of the function to be applied. This may take some refactoring of the elementwise operations. (It *may* be OK, as all the others get "fused away", similarly for DateTime, but we should probably add a check that we don't try to partition anything but frames).

Reply:
Thank you!

In order to avoid refactoring the elementwise methods I made DeferredStringMethods store two expressions, one that accesses Series.str, and one for the underlying Series. This seems to get things passing, but I'm not 100% sure this is safe. I'll look at extending `elementwise_method` for this use-case.

Reply:
Done